Source/WebKit/Platform/IPC/cocoa/SharedFileHandleCocoa.cpp

```diff
@@ -41,27 +41,21 @@ std::optional<SharedFileHandle> SharedFileHandle::create(FileSystem::PlatformFil
     return SharedFileHandle { WTFMove(handle) };
 }
 
-void SharedFileHandle::encode(Encoder& encoder) const
+SharedFileHandle::SharedFileHandle(MachSendRight&& fileport)
 {
-    mach_port_name_t fileport = MACH_PORT_NULL;
-    if (fileport_makeport(m_handle.handle(), &fileport) == -1) {
+    int fd = fileport_makefd(fileport.sendRight());
+    if (fd == -1)
         return;
-    }
-
-    encoder << MachSendRight::adopt(fileport);
+    m_handle = WebCore::FileHandle { WTFMove(fd) };
```

Comment:
This taking a PlatformFileHandle&& is a little strange since all the PlatformFileHandle types fit in a register.